lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -500,6 +500,9 @@ def get_thread_list(
         topic_id_list=None,
         text_search=None,
         following=False,
+        filter_own_threads=None,
```

Comment:
Instead of this, we should call this `author`, and make sure the requesting user and `author` are the same unless the user is a moderator. Although since this is a filter on existing data, I guess it's safe to just allow filtering by any user.

Reply:
NOTE: make sure filtering by non-existing user doesn't leak usernames.